ts/Stock/Indicators/Stochastic/StochasticIndicator.ts

```diff
@@ -205,11 +209,28 @@ class StochasticIndicator extends SMAIndicator {
             HL = extremes[1] - LL;
             K = CL / HL * 100;
 
+            if (isNaN(K) && constantValues) {
+                j++;
+                continue;
+            } else if (constantValues && !isNaN(K)) {
+                constantValues = false;
+            }
+
             xData.push(xVal[i]);
-            yData.push([K, null]);
+
+            if (isNaN(K)) {
+                yData.push([
+                    yData[yData.length - 1] &&
```

Comment:
Micro opt: cache manually the variable. In some cases browsers nicely cache the value, however, I'm not sure about this particular case - we actually do change the length of the array inside the if-else block. You can get the value easily from `push()`: `length = yData.push( ... );` and use it: `yData[length - 1]` or even better: `length = yData.push( ... ) - 1;` (however, personally, I don't like this hanging `-1` style). Or use it in a line above when pushing to `xData`

Note: I couldn't find how`.length` getter works - [ecma](https://262.ecma-international.org/13.0/#sec-properties-of-array-instances-length) is enigmatic. Does it actually look for the last object on the array, then read the index, then increment it?